### ISSUE TITLE
Error 500 'asset_maintenances' page

### ISSUE THREAD

**Issue by Mynty100** [grokability/snipe-it#12934]
### Debug mode

- [X] I have enabled debug mode
- [X] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

### Describe the bug

When trying to got to the asset_maintenances page the site gives Error 500.
Going to an asset and clicking at maintenance tap works and loads correct data.
Have run >php upgrade.php been doing this all of version 6 SnipeIT.

### Reproduction steps

Got to 'Reports -> asset maintenances reports'

### Expected behavior

Page to load and show all maintenances log on all devices.

### Screenshots

![image](https://user-images.githubusercontent.com/34461276/234750367-86194dec-33cb-4012-95c3-163cc1f49f67.png)

### Snipe-IT Version

6.1.0

### Operating System

Ubuntu

### Web Server

Apache

### PHP Version

7.4

### Operating System

Windows 11

### Browser

Firefox and Edge

### Version

any

### Error messages

```shell
With debug turned on just white screen / error 500.
Logs show nothing of value. 
php -m 
[PHP Modules]
bcmath
calendar
Core
ctype
curl
date
dom
exif
FFI
fileinfo
filter
ftp
gd
gettext
hash
iconv
json
ldap
libxml
mbstring
mysqli
mysqlnd
openssl
pcntl
pcre
PDO
pdo_mysql
Phar
posix
readline
Reflection
session
shmop
SimpleXML
sockets
sodium
SPL
standard
sysvmsg
sysvsem
sysvshm
tokenizer
xml
xmlreader
xmlwriter
xsl
Zend OPcache
zip
zlib

[Zend Modules]
Zend OPcache
```

**Comment by marcusmoore** [comment]
Hi @Mynty100, let's see if we can figure this out. 

> Logs show nothing of value. 

Do you mean browser's console logs or application log? (Application log is usually located in `storage/logs/laravel.log`)

Just to make sure: the other report pages, Activity Report, Audit Log, etc, are displayed just fine?

**Comment by Mynty100** [comment]
Hi @marcusmoore 

Thanks for getting back to me. I have opened the laravel.log. Please see the below info from yesterday while site was in 'debug' mode.

______________________________________________________________________________________________________________________________________
[2023-04-27 12:30:34] production.ERROR: Symfony\Component\ErrorHandler\Error\FatalError: Allowed memory size of 134217728 bytes exhausted (tried to allocate 2097160 bytes) in /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Relations/BelongsTo.php:134
Stack trace:
#0 {main}  
[2023-04-27 12:30:34] production.ERROR: Allowed memory size of 134217728 bytes exhausted (tried to allocate 2097160 bytes) {"userId":1358,"exception":"[object] (Symfony\\Component\\ErrorHandler\\Error\\FatalError(code: 0): Allowed memory size of 134217728 bytes exhausted (tried to allocate 2097160 bytes) at /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Relations/BelongsTo.php:134)
[stacktrace]
#0 {main}
"} 
[2023-04-27 12:33:21] production.ERROR: Symfony\Component\ErrorHandler\Error\FatalError: Allowed memory size of 134217728 bytes exhausted (tried to allocate 2097160 bytes) in /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Relations/BelongsTo.php:134
Stack trace:
#0 {main}  
[2023-04-27 12:33:21] production.ERROR: Allowed memory size of 134217728 bytes exhausted (tried to allocate 2097160 bytes) {"userId":1358,"exception":"[object] (Symfony\\Component\\ErrorHandler\\Error\\FatalError(code: 0): Allowed memory size of 134217728 bytes exhausted (tried to allocate 2097160 bytes) at /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Relations/BelongsTo.php:134)
[stacktrace]
#0 {main}
"} 
[2023-04-27 12:33:23] production.ERROR: Symfony\Component\ErrorHandler\Error\FatalError: Allowed memory size of 134217728 bytes exhausted (tried to allocate 12288 bytes) in /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php:496
Stack trace:
#0 {main}  
[2023-04-27 12:33:23] production.ERROR: Allowed memory size of 134217728 bytes exhausted (tried to allocate 12288 bytes) {"userId":1358,"exception":"[object] (Symfony\\Component\\ErrorHandler\\Error\\FatalError(code: 0): Allowed memory size of 134217728 bytes exhausted (tried to allocate 12288 bytes) at /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php:496)
[stacktrace]
#0 {main}
"} 
[2023-04-27 12:33:26] production.ERROR: Symfony\Component\ErrorHandler\Error\FatalError: Allowed memory size of 134217728 bytes exhausted (tried to allocate 12288 bytes) in /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php:496
Stack trace:
#0 {main}  
[2023-04-27 12:33:26] production.ERROR: Allowed memory size of 134217728 bytes exhausted (tried to allocate 12288 bytes) {"userId":1358,"exception":"[object] (Symfony\\Component\\ErrorHandler\\Error\\FatalError(code: 0): Allowed memory size of 134217728 bytes exhausted (tried to allocate 12288 bytes) at /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php:496)
[stacktrace]
#0 {main}
"} 
______________________________________________________________________________________________________________________________________

All other report pages working as normal.

Regards,
Mynty.

**Comment by marcusmoore** [comment]
Thanks for the details. 

To render that view it looks like the controller [loads _all_ asset maintenances](https://github.com/snipe/snipe-it/blob/67471a3a775c49f488201be4025e7e0bea3e9feb/app/Http/Controllers/ReportsController.php#L902C13-L905) (and I'm guessing you have a lot) which is causing PHP to run out of memory.

Interestingly, the view that gets rendered doesn't even use the models that were loaded and [instead calls the api](https://github.com/snipe/snipe-it/blob/ecb11c3a12cca02b8ccc82657b3edfbb9afad8c6/resources/views/reports/asset_maintenances.blade.php#L30) for them and that method [limits the amount of eloquent models](https://github.com/snipe/snipe-it/blob/22d73f503a610db04fb5fa1bc614a389d4f0f5f0/app/Http/Controllers/Api/AssetMaintenancesController.php#L98) it loads into memory which would avoid the issue you're experiencing.

I'm going to push a PR to fix this soon.

**Comment by Mynty100** [comment]
Cheers @marcusmoore 

yes i have a powershell script that api calls every time a user logs in for auto check out and maintenance logs details about IP address, OS version, who logged in and if the user that logged it is the assigned user or not. Think last time saw this working there was over 54000 records.

**Comment by marcusmoore** [comment]
@Mynty100 a fix was just merged to develop (#12943). You could use the upgrade script and pass in "develop" to pull that branch or wait a bit for the next release to be tagged. We're hoping to get that out fairly soon.

**Comment by Mynty100** [comment]
Hi @marcusmoore Thanks for that. After you pointed out was a memory issue, as a temp work around I've set the php.ini memory to -1 allowing it to use however much it needs. Server has 8gb RAM and only using 2gb at any given time. Seems to be now allowing the site to run but will change it back once next release is out. Thank you for you hard work and communication regarding this. Appreciate it.
